prow/plugins/label-based-commenting/label-based-commenting.go

```diff
@@ -0,0 +1,286 @@
+	if ce.IssueState != "open" || ce.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	cp, err := pc.CommentPruner()
+	if err != nil {
+		return err
+	}
+
+	return handleComment(pc.Logger, pc.GitHubClient, cp, pc.PluginConfig.LabelBasedCommenting, &ce)
+}
+
+func handleComment(log *logrus.Entry, ghc rblm.GithubClient, cp rblm.CommentPruner, configs []plugins.LabelBasedCommenting, e *github.GenericCommentEvent) error {
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	number := e.Number
+
+	event := &rblm.Event{
+		Org:    org,
+		Repo:   repo,
+		Number: number,
+		Author: e.User.Login,
+		IsPR:   e.IsPR,
+	}
+	if e.IsPR {
+		pr, err := ghc.GetPullRequest(org, repo, number)
+		if err != nil {
+			return err
+		}
+		event.Branch = pr.Base.Ref
+	}
+	return handle(log, ghc, cp, configs, event)
+}
+
+func handle(log *logrus.Entry, ghc rblm.GithubClient, cp rblm.CommentPruner, configs []plugins.LabelBasedCommenting, e *rblm.Event) error {
+	// Find any configs that may be relevant to this event.
+	matchConfigs := matchingConfigs(e.Org, e.Repo, e.Branch, e.Label, configs)
+	if len(matchConfigs) == 0 {
+		return nil
+	}
+
+	err := rblm.LabelPreChecks(e, ghc, extractRegexBasedLabelMatchConfigs(configs))
+	if err != nil {
+		return err
+	}
+
+	// Handle the potentially relevant configs.
+	for _, cfg := range matchConfigs {
+		hasMatchingLabel := false
+		for _, label := range e.CurrentLabels {
+			hasMatchingLabel = hasMatchingLabel || cfg.Re.MatchString(label.Name)
+		}
+
+		commentResponse := plugins.FormatSimpleResponse(e.Author, constructCommentResponse(e, cfg.Comment))
+		if hasMatchingLabel {
+			// TODO(MadhavJivrajani): we need to check if the comment has already
+			// been applied.
+			if err := ghc.CreateComment(e.Org, e.Repo, e.Number, commentResponse); err != nil {
+				log.WithError(err).Error("Failed to create matching comment.")
+			}
+		} else {
+			cp.PruneComments(func(comment github.IssueComment) bool {
+				return strings.Contains(comment.Body, commentResponse)
+			})
+		}
+
+	}
+	return nil
+}
+
+func extractRegexBasedLabelMatchConfigs(configs []plugins.LabelBasedCommenting) []plugins.RegexBasedLabelMatch {
+	res := make([]plugins.RegexBasedLabelMatch, 0, len(configs))
+	for _, cfg := range configs {
+		res = append(res, cfg.RegexBasedLabelMatch)
+	}
+
+	return res
+}
+
+// constructCommentResponse constructs the comment response based on
+// the user provided comment. If the repository in question (obtained
+// through e) is kubernetes/kubernetes and the comment is being posted
+// in response to an event on an *issue*, and it is a label event with
+// the label being kind/support, the comment response will have a list
+// of relevant communication media present as well (based on the list
+// of the current labels). In all other cases, the comment is returned
+// as is.
+func constructCommentResponse(e *rblm.Event, comment string) string {
+	if !isRepoKubernetesKubernetes(e.Org, e.Repo) ||
```

Comment:
I don't love this repo having a hard-coded exemption. Can we make this configurable instead